compute_tools/src/http/api.rs

```diff
@@ -98,6 +98,32 @@ async fn routes(req: Request<Body>, compute: &Arc<ComputeNode>) -> Response<Body
             }
         }
 
+        (&Method::POST, "/extensions") => {
+            info!("serving /extensions POST request");
+            let status = compute.get_status();
+            if status != ComputeStatus::Running {
+                let msg = format!(
+                    "invalid compute status for extensions request: {:?}",
+                    status
+                );
+                error!(msg);
+                return Response::new(Body::from(msg));
+            }
+
+            let body = hyper::body::to_bytes(req.into_body()).await.unwrap();
+            let body = serde_json::from_slice::<serde_json::Value>(&body).unwrap();
+            let extension = body["extension"].as_str().unwrap();
+            let database = body["database"].as_str().unwrap();
+            let res = compute.install_extension(extension, database);
```

Comment:
Does this need to be async?

Reply:
turns out that it does - runtime inside a runtime problems.